### ISSUE TITLE
Singleton VS new Object

### ISSUE THREAD

**Issue by usama33** [jaggedsoft/node-binance-api#81]
Hi,

I just realized that api is singleton. What If user wants to create multiple binance api objects based on different api key/secrets? I did few changes and it works now with 'new'. If you want I can create a pull request for you or following is what I did;

Replaced this;
`module.exports = function() {`

with
`function Binance() {
	if (!(this instanceof Binance)) {
    return new Binance();
  }`

And added
`module.exports = Binance;`

Usage
`var Binance = require('node-binance-api');`
`module.exports = class MyWrapper {`
`constructor() { this.binance = new Binance(); }`
`}`

**Comment by jaggedsoft** [comment]
Thank you for doing this! I'm assuming this is a breaking change? If so, I will schedule this for a release soon at version `v0.5.0`

**Comment by springjben** [comment]
```module.exports = class MyWrapper {
        constructor() { this.binance = new Binance(); }
      }```

With the above statement, how do I access the binance information outside of this?

**Comment by usama33** [comment]
It doesn't necessarily mean you should be creating inside the class..

var myBinanceObject = new Binance();
myBinanceObject.options({
      'APIKEY': apiKey,
      'APISECRET': apiSecret,
      useServerTime: true
    }, null);

above should work as well... If you wanna call from within class... Then in
any function of the class you can call with "this.binance"

Regards
*Muhammad Usama*
*Mob: 206 698 5322*

*** IMPORTANT DISCLAIMER ***
This email, including any attachments, is intended only for use by the
addressee(s) and may contain confidential and/or personal information. Any
personal information contained in this email is not to be used or disclosed
for any purpose other than the purpose for which you have received it.

If you are not the intended recipient you are notified that any review,
re-transmission, disclosure, use or dissemination of this communication is
strictly prohibited by Australian State and Commonwealth Laws including the
Victorian Information Privacy Act 2000. If you have received this
communication in error please contact the sender immediately and delete all
copies of this transmission together with any attachments.

**Comment by jaggedsoft** [comment]
@dmzoneill I am having trouble implementing this, but since we just renamed `trades` to `aggTrades` now would be a great time to bump the version number up to 6.0. Would really appreciate your help with this if you have time

**Comment by dmzoneill** [comment]
Cool, im not awesomely familiar with nodejs... more of a web/jquery guy.. 

It cant be that hard to figure out.. ill see if i can find an hour over the weekend..

 the misses quit her job.. shes around and taking up my time atm..... 😉

**Comment by Ionut-Milas** [comment]
hi @jaggedsoft Is the deployed?
I'm getting a "Binance is not a constructor" error.

**Comment by jaggedsoft** [comment]
@Ionut-Milas No, I have attempted this, but I am unable to get it to work as desired. Someone else will need to do it, or I will revisit it later

I want this feature

**Comment by jaggedsoft** [comment]
@usama33 Fantastic mate, that is very helpful. Thank you!! I'll issue a new release and give you full credit

**Comment by jaggedsoft** [comment]
Available in v0.7.0 https://github.com/jaggedsoft/node-binance-api/commit/57706f636c02df02b50cb0ab67f257183cdfc6ac
Thank you again!!

**Comment by bmino** [comment]
As far as I can tell, the instances are working great!
However, does this break the ability to share the internal depth cache?

For instance, let's say an application (using node-binance-api as a dependency) opens a few websocket streams using the depthCache() method which populates the internal depthCache object. Now, two parts of this application want to require node-binance-api and read from the depthCache. Is this no longer possible without maintaining another local copy of the depthCache in the application?

**Comment by jaggedsoft** [comment]
@bmino I'm not sure the best way to do this. What I had considered doing in the past was packaging all websocket functions in their own module. Since only account balance and trade execution updates require a key

**Comment by bmino** [comment]
What if the following block of code is removed? It forces instantiation.
```
if (!(this instanceof Binance)) {
    return new Binance();
}
```

Without that block, I think you can create both instances and singletons like this:
```
let api = require('node-binance-api');
let instance = new api();
let singleton = api();
```

Here is a gist of my local simplified example testing this:
https://gist.github.com/bmino/a52998e6c184437ac0e524cf8ff06daa

**Comment by bmino** [comment]
I submitted PR #235 which will revert the default behavior of 
```javascript
const api = require('node-binance-api');
```
to return a singleton, but users can create an instance by the steps I mentioned above. Can someone test this with the newer `import` syntax?

**Comment by bmino** [comment]
There are some subtleties I missed in simplifying my gist test case. Mainly the use of assigning all variables to the "this" scope of the function export. I am now testing with the entirely of the library locally (instead of a simplified version) to ensure I don't miss anything.

But, it appears that the api credentials used in the test script are invalid now?

**Comment by dmzoneill** [comment]
Yes binance invalidated all keys , im on business travel till sunday.  Not sure if will get to it before then

@jaggedsoft

**Comment by jaggedsoft** [comment]
I don't have a test account to issue new sandbox keys, sorry. I suppose we will have to be patient :) Thanks guys

**Comment by bmino** [comment]
I submitted PR #245 to address the issues of shared variables between singletons.

Testing:
* I made a few attempts to fix the unit tests but to do that right will take much more time. The errors encountered are all due to an invalid key being used as mentioned in #241 
* I tested it in my project which restores it to full functionality (using the singleton pattern).
* I also tested it against this gist https://gist.github.com/bmino/bbb4669c7c40947166c9b1fc1d05ad8a
